File: api/src/main/java/put/poznan/sport/entity/SportEquipment.java
package put.poznan.sport.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import put.poznan.sport.entity.sportFacility.SportFacility;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class SportEquipment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String type;

    @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private String model;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private String imageUrl;

    @ManyToOne
    @JsonIgnore
    private SportFacility ownerSportFacility;

    @Column(nullable = false)
    private Integer amount;

}

File: mobile_app/lib/models/details/sport_equipment.dart
class SportEquipment {
  int? id;
  String? type;
  String? brand;
  String? model;
  String? description;
  String? imageUrl;
  int? amount;

  SportEquipment({
    this.id,
    this.type,
    this.brand,
    this.model,
    this.description,
    this.imageUrl,
    this.amount,
  });

  factory SportEquipment.fromJson(Map<String, dynamic> json) {
    return SportEquipment(
      id: json['id'],
      type: json['type'],
      brand: json['brand'],
      model: json['model'],
      description: json['description'],
      imageUrl: json['imageUrl'],
      amount: json['amount'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'type': type,
      'brand': brand,
      'model': model,
      'description': description,
      'imageUrl': imageUrl,
      'amount': amount,
    };
  }
}
